lib/rules/boolean-prop-naming.js

```diff
@@ -0,0 +1,238 @@
+/**
+ * @fileoverview Enforces consistent naming for boolean props
+ * @author Evgueni Naverniouk
+ */
+'use strict';
+
+const has = require('has');
+const Components = require('../util/Components');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      category: 'Stylistic Issues',
+      description: 'Enforces consistent naming for boolean props',
+      recommended: false
+    },
+
+    schema: [{
+      additionalProperties: false,
+      properties: {
+        propTypeNames: {
+          items: {
+            type: 'string'
+          },
+          type: 'array'
+        },
+        rule: {
+          default: null,
+          type: 'string'
+        }
+      },
+      type: 'object'
+    }]
+  },
+
+  create: Components.detect(function(context, components, utils) {
+    const DEFAULT_RULE = '^(is|has)[A-Z]([A-Za-z0-9]?)+';
+    const sourceCode = context.getSourceCode();
+    const config = context.options[0] || {};
+    const rule = new RegExp(config.rule || DEFAULT_RULE);
+    const propTypeNames = config.propTypeNames ? config.propTypeNames : ['bool'];
```

Comment:
For some reason, `default` schema definition doesn't work for this option. Maybe because it's an `array` type? Specifying a `default` at the schema level here never returns anything from config object.